PR #21 Reformatted (#128)
mongodb/docs-meta: source/style-guide/markup/format/substitutions.txt

=============
Substitutions
=============

> Source Constants

Reviewer: [issue]

This seems like it needs some kind of introductory sentence or paragraph before the Source Constants heading?

Author: Good catch. I think it would serve the reader best to keep the title "Substitutions" since "Source Constants" might not be trivial to understand if not already familiar with it.

----------------

Source constants define substitution text for variables in {+rst+} files.
When building the documentation, the substitution text replaces the instances
of the variables prior to {+docutils+} parser transforming the document to

> HTML. Source constants can be used within directives and roles.

Reviewer 2: ```suggestion
HTML. You can use source constants within directives and roles.
```

You can add or modify source constants in your documentation repository's
``snooty.toml`` file in the base directory. Source constants should be
defined in the ``[constants]`` section using in the
``<name> = "substitution text"`` format.

For example, suppose you want to add a source constant that represents the
URL to the MDN documentation, "https://developer.mozilla.org/en-US/docs/MDN/".
If you name the source constant "mdn", your ``snooty.toml`` file should
include the following line in the ``[constants]`` section:

.. code-block:: python

   mdn = "https://developer.mozilla.org/en-US/docs/MDN/"

> To use the source constant, as shown in the following example:

Reviewer 2: ```suggestion
Use the source constant as shown in the following example:
```

.. code-block:: rst

   Read the `Game development <{ +mdn+ }Games>`__ article on MDN for information
   on creating web games.

.. important::

>    In your {+rst+}, source constants should not contain spaces between curly

Reviewer 2: **Style:** See [Contractions](https://www.mongodb.com/docs/meta/style-guide/style/contractions/).

```suggestion
   In your {+rst+}, source constants shouldn't contain spaces between curly
```

Substitutions
=============

In Snooty, you can use variables to represent a text value in the
rendered form of the documentation. These variables are called
source constants. When used consistently, they help prevent inconsistencies in
formatting the term and make it easy to change the value for all its

> occurrences.

Reviewer: [suggestion]

To help contextualize this, it might help to briefly mention a use case for substitutions. Consider adding something like:

"For example, you can use a substitution for the name of a MongoDB product, like Compass. If the product name changes, you only need to update the variable value to update all instances of "Compass" in the entire documentation site."
